src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -2406,58 +2427,92 @@ def add_dns_delegation(cmd, child_zone, parent_zone, child_rg, child_zone_name):
+            registration_vnets_dict.append({"id": registration_vnet.id})
+
+    created_zone = _DNSZoneCreate(cli_ctx=cmd.cli_ctx)(command_args={
+        "resource_group": resource_group_name,
+        "zone_name": zone_name,
+        "if_none_match": '*' if if_none_match else None,
+        "location": 'global',
+        "tags": tags,
+        "zone_type": zone_type,
+        "resolution_virtual_networks": resolution_vnets_dict if resolution_vnets else None,
+        "registration_virtual_networks": registration_vnets_dict if registration_vnets else None
+    })
 
     if cmd.supported_api_version(min_api='2016-04-01', resource_type=ResourceType.MGMT_NETWORK_DNS) and parent_zone_name is not None:
         logger.info('Attempting to add delegation in the parent zone')
         add_dns_delegation(cmd, created_zone, parent_zone_name, resource_group_name, zone_name)
     return created_zone
 
 
-def update_dns_zone(instance, tags=None, zone_type=None, resolution_vnets=None, registration_vnets=None):
+def update_dns_zone(cmd, resource_group_name, zone_name, tags=None, zone_type=None, resolution_vnets=None, registration_vnets=None, if_match=None):
+
+    zone = {
+        "resource_group": resource_group_name,
+        "zone_name": zone_name,
+        "if_match": if_match,
+    }
 
     if tags is not None:
-        instance.tags = tags
+        zone["tags"] = tags
 
     if zone_type:
-        instance.zone_type = zone_type
+        zone["zone_type"] = zone_type
 
     if resolution_vnets == ['']:
-        instance.resolution_virtual_networks = None
+        zone["resolution_virtual_networks"] = None
     elif resolution_vnets:
-        instance.resolution_virtual_networks = resolution_vnets
+        resolution_vnets_dict = []
+        for resolution_vnet in resolution_vnets:
+            resolution_vnets_dict.append({"id": resolution_vnet.id})
+        zone["resolution_virtual_networks"] = resolution_vnets_dict
 
     if registration_vnets == ['']:
-        instance.registration_virtual_networks = None
+        zone["registration_virtual_networks"] = None
     elif registration_vnets:
-        instance.registration_virtual_networks = registration_vnets
-    return instance
+        registration_vnets_dict = []
+        for registration_vnet in registration_vnets:
+            registration_vnets_dict.append({"id": registration_vnet.id})
+        zone["registration_virtual_networks"] = registration_vnets_dict
+
+    return _DNSZoneUpdate(cli_ctx=cmd.cli_ctx)(command_args=zone)
 
 
 def list_dns_zones(cmd, resource_group_name=None):
-    ncf = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_NETWORK_DNS).zones
     if resource_group_name:
-        return ncf.list_by_resource_group(resource_group_name)
-    return ncf.list()
+        return _DNSZoneListByRg(cli_ctx=cmd.cli_ctx)(command_args={
+            "resource_group": resource_group_name
+        })
+
+    return _DNSZoneListBySub(cli_ctx=cmd.cli_ctx)(command_args={})
```

Comment:
Usually we will combine `list-by-resource-group` with `list` by subscription. You can follow this doc [link](https://azure.github.io/aaz-dev-tools/pages/usage/workspace-editor/#merge-similar-resource-urls-in-one-command)

Reply:
We cannot combine the list calls because AAZ tool creates separate commands under separate groups.

Reply:
Merged the list APIs into one command